tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py

```diff
@@ -0,0 +1,287 @@
+        )
+        if context[f"{cronjob_name}_jobs"]:
+            return
+        time.sleep(10)
+    assert False, f"CronJob {cronjob_name} has no jobs after {timeout} seconds"
+
+
+# }}}
+# When {{{
+
+
+@when(
+    parsers.parse("we list the '{state}' CronJobs"),
+)
+def call_metalk8s_kubernetes_cronjob_get_cronjobs(host, ssh_config, context, state):
+    command = [
+        "salt-run",
+        "salt.cmd",
+        "metalk8s_kubernetes_cronjob.get_cronjobs",
+        "namespace=default",
+        "--out json",
+        "--log-level quiet",
+    ]
+    if state not in ["active", "suspended"]:
+        raise ValueError(f"Invalid silence state: {state}")
+    # True if suspended, false if active
+    command.append(f"suspended={state == 'suspended'}")
+    result = run_salt_command(
+        host,
+        command,
+        ssh_config,
+    )
+    context[f"cronjobs_{state}"] = json.loads(result.stdout)
+
+
+@when(
+    parsers.parse(
+        "we suspend the '{cronjob_name}' CronJob marking it with '{suspend_mark}'",
+    ),
+)
+def call_metalk8s_kubernetes_cronjob_suspend_cronjob(
+    host,
+    ssh_config,
+    cronjob_name,
+    suspend_mark,
+):
+    command = [
+        "salt-run",
+        "salt.cmd",
+        "metalk8s_kubernetes_cronjob.suspend_cronjob",
+        f"name={cronjob_name}",
+        "namespace=default",
+        f"mark={suspend_mark}",
+        "--out json",
+        "--log-level quiet",
+    ]
+    result = run_salt_command(
+        host,
+        command,
+        ssh_config,
+    )
+    assert json.loads(result.stdout) is True, "Failed to suspend the CronJob"
+
+
+@when(
+    parsers.parse("we activate the '{cronjob_name}' Cronjob"),
+)
+def call_metalk8s_kubernetes_cronjob_activate_cronjob(host, ssh_config, cronjob_name):
+    command = [
+        "salt-run",
+        "salt.cmd",
+        "metalk8s_kubernetes_cronjob.activate_cronjob",
+        f"name={cronjob_name}",
+        "namespace=default",
+        "--out json",
+        "--log-level quiet",
+    ]
+    result = run_salt_command(
+        host,
+        command,
+        ssh_config,
+    )
+    assert json.loads(result.stdout) == True, "Failed to activate the CronJob"
+
+
+@when(
+    parsers.parse(
+        "we stop the '{cronjob_name}' CronJob marked with '{suspend_mark}' "
+        "and delete all its Jobs"
+    ),
+)
+def call_metalk8s_kubernetes_cronjob_suspend_cronjob_and_delete_jobs(
+    host, ssh_config, context, cronjob_name, suspend_mark
+):
+    command = [
+        "salt-run",
+        "salt.cmd",
+        "metalk8s_kubernetes_cronjob.suspend_cronjob_and_delete_jobs",
+        f"name={cronjob_name}",
+        "namespace=default",
+        f"mark={suspend_mark}",
+        "wait=true",
+        "--out json",
+        "--log-level quiet",
+    ]
+
+    result = run_salt_command(
+        host,
+        command,
+        ssh_config,
+    )
+    context[f"deleted_jobs"] = json.loads(result.stdout)
```

Comment:
```suggestion
    context["deleted_jobs"] = json.loads(result.stdout)
```

Edit: This one does not seems used btw

Reply:
Fixed.